invalid escape sequence: `\U`

It seems like there is a problem when you are trying to deploy a contract on Windows (client side) and it seems related to the way the software is handling the slashes and backslashes in the paths. I'm getting this error:

```
PS C:\Users\alexa\.eris\apps\idi> eris contracts  test --dir=C:/Users/alexa/.eris/apps/idi
panic: regexp: Compile(`eris_data_C:\Users\alexa\.eris\apps\idi_88fcbf03_1`): error parsing regexp: invalid escape sequence: `\U`

goroutine 1 [running]:
regexp.MustCompile(0xc082135dc0, 0x32, 0x5)
        c:/go/src/regexp/regexp.go:221 +0x176
github.com/eris-ltd/eris-cli/util.ParseContainers(0xc082135dc0, 0x32, 0xc081feaf01, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        D:/Go/src/github.com/eris-ltd/eris-cli/util/container_stats.go:293 +0x216
github.com/eris-ltd/eris-cli/perform.ContainerExists(0xc08207d000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        D:/Go/src/github.com/eris-ltd/eris-cli/perform/docker_run.go:625 +0x5a
github.com/eris-ltd/eris-cli/perform.DockerCreateData(0xc08207d000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/perform/docker_run.go:38 +0x14a
github.com/eris-ltd/eris-cli/chains.setupChain(0xc0820ec000, 0xa4b438, 0x3, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/chains/operate.go:244 +0x2be
github.com/eris-ltd/eris-cli/chains.NewChain(0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/chains/operate.go:44 +0x4c1
github.com/eris-ltd/eris-cli/chains.ThrowAwayChain(0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/chains/operate.go:92 +0x3c8
github.com/eris-ltd/eris-cli/contracts.bootThrowAwayChain(0xc08221a800, 0x1d, 0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/contracts/operate.go:302 +0xa6
github.com/eris-ltd/eris-cli/contracts.BootServicesAndChain(0xc0820ec000, 0xc082238000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/contracts/operate.go:85 +0x469
github.com/eris-ltd/eris-cli/contracts.RunPackage(0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/contracts/operate.go:37 +0x42e
github.com/eris-ltd/eris-cli/commands.ContractsTest(0xd40ce0, 0xc0821003d0, 0x0, 0x1)
        D:/Go/src/github.com/eris-ltd/eris-cli/commands/contracts.go:149 +0x114
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).execute(0xd40ce0, 0xc082100380, 0x1, 0x1, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:530 +0x870
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).Execute(0xd41e60, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:613 +0x59a
github.com/eris-ltd/eris-cli/commands.Execute()
        D:/Go/src/github.com/eris-ltd/eris-cli/commands/eris.go:66 +0x39
main.main()
        D:/Go/src/github.com/eris-ltd/eris-cli/cmd/eris/main.go:8 +0x1b

goroutine 5 [select]:
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.readLoop()
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger.go:116 +0x202
created by github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.init.1
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger.go:28 +0x32

goroutine 6 [syscall]:
os/signal.loop()
        c:/go/src/os/signal/signal_unix.go:22 +0x1f
created by os/signal.init.1
        c:/go/src/os/signal/signal_unix.go:28 +0x3e

goroutine 18 [IO wait]:
net.runtime_pollWait(0x1295be0, 0x72, 0xc082143750)
        c:/go/src/runtime/netpoll.go:157 +0x67
net.(*pollDesc).Wait(0xc082021070, 0x72, 0x0, 0x0)
        c:/go/src/net/fd_poll_runtime.go:73 +0x41
net.(*ioSrv).ExecIO(0xc08202a1c0, 0xc082020f60, 0xa47b10, 0x7, 0xb4b450, 0x101d, 0x0, 0x0)
        c:/go/src/net/fd_windows.go:182 +0x177
net.(*netFD).Read(0xc082020f00, 0xc08224a000, 0x2000, 0x2000, 0x0, 0x0, 0x0)
        c:/go/src/net/fd_windows.go:463 +0x17e
net.(*conn).Read(0xc082148000, 0xc08224a000, 0x2000, 0x2000, 0x0, 0x0, 0x0)
        c:/go/src/net/net.go:172 +0xeb
crypto/tls.(*block).readFromUntil(0xc082146870, 0x14a0000, 0xc082148000, 0x5, 0x0, 0x0)
        c:/go/src/crypto/tls/conn.go:455 +0xd3
crypto/tls.(*Conn).readRecord(0xc08214c000, 0xb4bb17, 0x0, 0x0)
        c:/go/src/crypto/tls/conn.go:540 +0x2d8
crypto/tls.(*Conn).Read(0xc08214c000, 0xc08215c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        c:/go/src/crypto/tls/conn.go:901 +0x16e
net/http.noteEOFReader.Read(0x14a4468, 0xc08214c000, 0xc08210eb58, 0xc08215c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        c:/go/src/net/http/transport.go:1370 +0x6e
net/http.(*noteEOFReader).Read(0xc0822044a0, 0xc08215c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        <autogenerated>:126 +0xd7
bufio.(*Reader).fill(0xc0821fd3e0)
        c:/go/src/bufio/bufio.go:97 +0x1f0
bufio.(*Reader).Peek(0xc0821fd3e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        c:/go/src/bufio/bufio.go:132 +0xd3
net/http.(*persistConn).readLoop(0xc08210eb00)
        c:/go/src/net/http/transport.go:876 +0xfe
created by net/http.(*Transport).dialConn
        c:/go/src/net/http/transport.go:685 +0xc7f

goroutine 19 [select]:
net/http.(*persistConn).writeLoop(0xc08210eb00)
        c:/go/src/net/http/transport.go:1009 +0x413
created by net/http.(*Transport).dialConn
        c:/go/src/net/http/transport.go:686 +0xca4
```

And I've seen the following StackOveflow thread that is seems related: http://stackoverflow.com/questions/13737370/cmake-error-invalid-escape-sequence-u

Versions: 
```
PS C:\Users\alexa\.eris\apps\idi> eris version
Eris CLI Version: 0.11.0
```

I'm not sure if I run all the commands but this is the output and it seems everything is working in the same way, so same error.

```
Eris CLI Version: 0.11.0
PS C:\Users\alexa\.eris\apps> eris update --branch winConFix
PS C:\Users\alexa\.eris\apps> eris contracts  test --dir=C:/Users/alexa/.eris/apps/idi
panic: regexp: Compile(`eris_data_C:\Users\alexa\.eris\apps_d12d37b4_1`): error parsing regexp: invalid escape sequence: `\U`

goroutine 1 [running]:
regexp.MustCompile(0xc0822568a0, 0x2e, 0x5)
        c:/go/src/regexp/regexp.go:221 +0x176
github.com/eris-ltd/eris-cli/util.ParseContainers(0xc0822568a0, 0x2e, 0xc081fed501, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        D:/Go/src/github.com/eris-ltd/eris-cli/util/container_stats.go:293 +0x216
github.com/eris-ltd/eris-cli/perform.ContainerExists(0xc08207d000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        D:/Go/src/github.com/eris-ltd/eris-cli/perform/docker_run.go:625 +0x5a
github.com/eris-ltd/eris-cli/perform.DockerCreateData(0xc08207d000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/perform/docker_run.go:38 +0x14a
github.com/eris-ltd/eris-cli/chains.setupChain(0xc0820ec000, 0xa4b438, 0x3, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/chains/operate.go:244 +0x2be
github.com/eris-ltd/eris-cli/chains.NewChain(0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/chains/operate.go:44 +0x4c1
github.com/eris-ltd/eris-cli/chains.ThrowAwayChain(0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/chains/operate.go:92 +0x3c8
github.com/eris-ltd/eris-cli/contracts.bootThrowAwayChain(0xc082218800, 0x19, 0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/contracts/operate.go:302 +0xa6
github.com/eris-ltd/eris-cli/contracts.BootServicesAndChain(0xc0820ec000, 0xc082236000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/contracts/operate.go:85 +0x469
github.com/eris-ltd/eris-cli/contracts.RunPackage(0xc0820ec000, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/contracts/operate.go:37 +0x42e
github.com/eris-ltd/eris-cli/commands.ContractsTest(0xd40ce0, 0xc0821003d0, 0x0, 0x1)
        D:/Go/src/github.com/eris-ltd/eris-cli/commands/contracts.go:149 +0x114
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).execute(0xd40ce0, 0xc082100380, 0x1, 0x1, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:530 +0x870
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).Execute(0xd41e60, 0x0, 0x0)
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:613 +0x59a
github.com/eris-ltd/eris-cli/commands.Execute()
        D:/Go/src/github.com/eris-ltd/eris-cli/commands/eris.go:66 +0x39
main.main()
        D:/Go/src/github.com/eris-ltd/eris-cli/cmd/eris/main.go:8 +0x1b

goroutine 5 [select]:
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.readLoop()
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger.go:116 +0x202
created by github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.init.1
        D:/Go/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger.go:28 +0x32

goroutine 6 [syscall]:
os/signal.loop()
        c:/go/src/os/signal/signal_unix.go:22 +0x1f
created by os/signal.init.1
        c:/go/src/os/signal/signal_unix.go:28 +0x3e
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,26 +3,22 @@ os: osx
 language: go
 
 env:
-- MACHINE_NAME=eris SWARM=osx-tests APIVERSION=latest
+- MACHINE_NAME=eris-test-osx-$TRAVIS_JOB_ID SWARM=osx-tests APIVERSION=latest
 
 install:
 - eval "$(gimme 1.5)"
 - brew update
-- brew install docker docker-machine
+- brew install docker docker-machine jq
 - echo "change the below when `docker-machine --version | cut -d ' ' -f 3` is at 0.5.4"
 - sudo bash -c "curl -sSL https://github.com/docker/machine/releases/download/v0.5.4/docker-machine_darwin-amd64 >/usr/local/bin/docker-machine"
 - sudo chmod +x /usr/local/bin/docker-machine
 
 script:
 - go install ./cmd/eris
-- docker-machine create --driver amazonec2 eris
-- tests/test_tool.sh
+- docker-machine create --driver amazonec2 eris-test-osx-$TRAVIS_JOB_ID && tests/test_tool.sh
 
-after_success:
-- docker-machine rm --force eris
-
-after_failure:
-- docker-machine rm --force eris
+after_script:
+- docker-machine rm --force eris-test-osx-$TRAVIS_JOB_ID
 
 notifications:
   email: false

--- a/Dockerfile
+++ b/Dockerfile
@@ -48,7 +48,7 @@ RUN curl -sSL -o $INSTALL_BASE/wrapdocker https://raw.githubusercontent.com/jpet
 RUN chmod +x $INSTALL_BASE/wrapdocker
 
 # DOCKER-MACHINE (for testing)
-ENV DOCKER_MACHINE_VERSION 0.4.0
+ENV DOCKER_MACHINE_VERSION 0.5.4
 RUN curl -sSL -o $INSTALL_BASE/docker-machine \
   https://github.com/docker/machine/releases/download/v$DOCKER_MACHINE_VERSION/docker-machine_linux-amd64 && \
   chmod +x $INSTALL_BASE/docker-machine

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,3 +1,40 @@
 version: 1.0.{build}-{branch}
-build:
-  verbosity: minimal
+clone_folder: C:\src\github.com\eris-ltd\eris-cli
+
+environment:
+  GOPATH: C:\
+  GOBIN: C:\bin
+  PATH: '%PATH%;C:\bin'
+  MACHINE_NAME: 'eris-test-win-%APPVEYOR_BUILD_ID%'
+
+build_script:
+- cmd: >-
+
+    choco install -yr msys2 > nul
+
+    choco install -yr docker jq
+
+    choco install -yr docker-machine -version 0.4.1
+
+    cd c:\src\github.com\eris-ltd\eris-cli\cmd\eris
+
+    go get
+
+    go install
+
+    docker-machine create --driver amazonec2 --amazonec2-access-key %AWS_ACCESS_KEY% --amazonec2-secret-key %AWS_SECRET_KEY% --amazonec2-region %AWS_REGION% --amazonec2-vpc-id %AWS_VPC_ID% --amazonec2-security-group %AWS_SECURITY_GROUP% eris-test-win-%APPVEYOR_BUILD_ID%
+
+test_script:
+- ps: >-
+
+    docker-machine.exe env --shell=powershell eris-test-win-"$env:APPVEYOR_BUILD_ID" | Invoke-Expression
+
+    cd c:\src\github.com\eris-ltd\eris-cli
+
+    bash tests\test_tool.sh
+
+on_finish:
+- cmd: >-
+    docker-machine kill eris-test-win-%APPVEYOR_BUILD_ID%
+
+    docker-machine rm --force eris-test-win-%APPVEYOR_BUILD_ID%

--- a/loaders/contracts.go
+++ b/loaders/contracts.go
@@ -3,7 +3,7 @@ package loaders
 import (
 	"fmt"
 	"os"
-	"path"
+	"path/filepath"
 	"regexp"
 
 	"github.com/eris-ltd/eris-cli/config"
@@ -72,7 +72,7 @@ func DefaultContractPackage() (*definitions.Contracts, error) {
 	// we don't catch the directory error here as it should be caught prior to
 	//   calling this function.
 	pwd, _ := os.Getwd()
-	app.Name = path.Base(pwd)
+	app.Name = filepath.Base(pwd)
 	app.ChainName = ""
 	app.TestType = "epm"
 	app.DeployType = "epm"

--- a/tests/test_tool.sh
+++ b/tests/test_tool.sh
@@ -71,7 +71,7 @@ announce() {
 
 connect() {
   echo "Starting Machine."
-  if [[ "$machine" != "eris" ]] # "eris" should be used when CI testing against OSX and Windows.
+  if [[ "$machine" != eris-test-osx* ]] || [[ "$machine" != eris-test-win* ]]
   then
     docker-machine start $machine 1>/dev/null
   fi
@@ -86,9 +86,26 @@ connect() {
     exit 1
   else
     echo "Machine Started."
+    sleep 15
     if [[ "$machine" != "eris" ]]
     then
-      docker-machine regenerate-certs -f $machine 2>/dev/null
+      n=0
+      until [ $n -ge 10 ]
+      do
+        sleep 3
+        docker-machine regenerate-certs -f $machine &>/dev/null && break
+        n=$[$n+1]
+      done
+      if [ $n -ge 10 ]
+      then
+        echo "There was an error connecting to the machine. Exiting test"
+        echo
+        test_exit=1
+        turn_off
+        report
+        cd $start
+        exit $test_exit
+      fi
     fi
   fi
   sleep 5
@@ -157,14 +174,6 @@ packagesToTest() {
   export ERIS_PULL_APPROVE="true"
   export ERIS_MIGRATE_APPROVE="true"
 
-  # The first run of tests expect ipfs to be running
-  eris services start ipfs
-  if [ $? -ne 0 ]; then return 1; fi
-  ERIS_IPFS_HOST="http://$(docker inspect --format='{{.NetworkSettings.IPAddress}}' eris_service_ipfs_1)"
-  if [ $? -ne 0 ]; then return 1; fi
-  export ERIS_IPFS_HOST
-  sleep 5 # give ipfs node time to boot
-
   # Start the first series of tests
   go test ./perform/...
   passed Perform
@@ -183,11 +192,6 @@ packagesToTest() {
   if [ $? -ne 0 ]; then return 1; fi
   go test ./keys/...
   passed Keys
-  if [ $? -ne 0 ]; then return 1; fi
-
-  # The second series of tests expects ipfs to not be running
-  eris services stop ipfs -frx
-  unset ERIS_IPFS_HOST
   if [ $? -ne 0 ]; then return 1; fi
 
   # Start the second series of tests
@@ -231,10 +235,8 @@ packagesToTest() {
 
 clear_stuff() {
   echo "Clearing images and containers."
-  set +e
   docker rm $(docker ps -a -q) &>/dev/null
   docker rmi -f $(docker images -q) &>/dev/null
-  set -e
   echo ""
 }
 
@@ -267,7 +269,6 @@ report() {
 # ---------------------------------------------------------------------------
 # Go!
 
-set -e
 echo "Hello! The marmots will begin testing now."
 if [[ "$machine" == "eris-test-local" ]]
 then
@@ -276,7 +277,6 @@ else
   announce
   connect
 fi
-set +e
 
 # Once machine is turned on, display docker information
 echo ""

--- a/testutils/testing_utils.go
+++ b/testutils/testing_utils.go
@@ -48,7 +48,7 @@ func TestsInit(testType string) (err error) {
 	do.Actions = true
 	do.Yes = true
 	if err := ini.Initialize(do); err != nil {
-		IfExit(fmt.Errorf("TRAGIC. Could not initialize the eris dir.\n"))
+		IfExit(fmt.Errorf("TRAGIC. Could not initialize the eris dir: %s.\n", err))
 	}
 
 	log.Info("Test init completed. Starting main test sequence now")

